### DISCUSSION TITLE
Upgrade Z-Push 2.6.4.beta1 to 2.7.1 - Fatal Error

### DISCUSSION THREAD

**Discussion by akbypajoma** [Z-Hub/Z-Push#54] [Q&A] [answered]
Hi Community,

when i upgrade from 2.6.4beta to 2.7.1 the Sync to our Outlook clients break. In the z-push-error.log i get following error code

**Fatal error: /usr/share/z-push/lib/utils/utils.php:1485 - Uncaught Error: Call to undefined function datefmt_create() in /usr/share/z-push/lib/utils/utils.php:1485**

We use the kopano community server, if it is important to help me with this error.

kopano-server-packages:
Installiert:           11.0.2.51.76cf85b89-0+877.1

I hope you can help me to fix this error.

kind regards

**Comment by matidau** [comment]
Hi @akbypajoma

Per the documentation php 7 should ship with this. I'm making the assumption that you're using php 7 as you are using Kopano.

https://www.php.net/manual/en/intldateformatter.create.php

What OS / packages are you using for PHP?

Cheers,
Mat

**Reply by akbypajoma** [reply]
Hi Mat,

thanks for your reply.

Here is the output of the php package we are using on our ubuntu 18.04 Server

php -v
PHP 7.2.34-39+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Jun  8 2023 15:20:15) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.34-39+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies

kind regards
ali

**Reply by matidau** [reply]
I would recommend using php 7.4 as the minimum version. As 18.04 is eol for general support you may need to use a ppa such as https://launchpad.net/~sergey-dryabzhinsky/+archive/ubuntu/php74

Cheers,
Mat

**Reply by akbypajoma** [reply]
that means, that php 7.4 will work with z-push 2.7.1?

**Reply by matidau** [reply]
I have tested Ubuntu 20.04 with php7.4 and it has worked. Php7.4 is the latest version of php that Kopano supported for the community edition and I will keep testing against this until it loses major distro support. 

You could also try installing 2.7.0, which didn't have the changes that are producing that error.

Cheers,
Mat

Edit: and just some extra thoughts what packages does apt list as installed? It is possible to have multiple or mismatches of php-cli, php-fpm, and php.

**Reply by fbartels** [reply]
> l support you may need to use a ppa

this in general will not work with Kopano, as the old Kopano relies on a php module that needs to be compiled against the abi of the given php release. And the Kopano builds only provide this for the native php version of a given release.

to use a newer php version, it is therefore required to also update the os.

The same limitation no longer applies to the new “Kopano Cloud”, but this is not available for use without a commercial subscription.

**Reply by akbypajoma** [reply]
Thank you for your help, i appreciate it. i have enabled php 7.4.

But i have one error and then i should have anything working. :)

when i try to "**z-push-admin -a list**" it comes to following error

PHP Warning:  PHP Startup: Unable to load dynamic library 'mapi.so' (tried: /usr/lib/php/20190902/mapi.so (/usr/lib/php/20190902/mapi.so: cannot open shared object file: No such file or directory), /usr/lib/php/20190902/mapi.so.so (/usr/lib/php/20190902/mapi.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'mapi.so' (tried: /usr/lib/php/20190902/mapi.so (/usr/lib/php/20190902/mapi.so: cannot open shared object file: No such file or directory), /usr/lib/php/20190902/mapi.so.so (/usr/lib/php/20190902/mapi.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0

FatalMisconfigurationException: PHP-MAPI extension is not available

**php -m**

PHP Warning:  PHP Startup: Unable to load dynamic library 'mapi.so' (tried: /usr/lib/php/20190902/mapi.so (/usr/lib/php/20190902/mapi.so: cannot open shared object file: No such file or directory), /usr/lib/php/20190902/mapi.so.so (/usr/lib/php/20190902/mapi.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'mapi.so' (tried: /usr/lib/php/20190902/mapi.so (/usr/lib/php/20190902/mapi.so: cannot open shared object file: No such file or directory), /usr/lib/php/20190902/mapi.so.so (/usr/lib/php/20190902/mapi.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
PHP 7.4.33 (cli) (built: Jun  8 2023 15:22:42) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.33, Copyright (c), by Zend Technologies

Have you an idea?

Thanks

**Comment by matidau** [comment, marked as answer]
Hi @akbypajoma,

> Fatal error: /usr/share/z-push/lib/utils/utils.php:1485 - Uncaught Error: Call to undefined function datefmt_create() in /usr/share/z-push/lib/utils/utils.php:1485

Going back to your original problem, I've worked through this and on Ubuntu 18.04 this is provided by the php7.2-intl package. `sudo apt install php7.2-intl` Should get you going.

I would still recommend upgrading your Kopano/Z-Push to Ubuntu 20.04 and making use of PHP 7.4. To use the the version of PHP as recommended by @fbartels (i.e. not the ppa) you will need to install the php7.4 package with  `sudo apt install php7.4-intl`

Cheers,
Mat
